base/gfspvgmgr/virtual_group_manager.go

```diff
@@ -82,15 +82,15 @@ func (picker *FreeStorageSizeWeightPicker) pickIndex() (uint32, error) {
 	return 0, fmt.Errorf("failed to pick weighted random index")
 }
 
-func (vgfm *virtualGroupFamilyManager) pickVirtualGroupFamily(filter *vgmgr.PickVGFFilter) (*vgmgr.VirtualGroupFamilyMeta, error) {
+func (vgfm *virtualGroupFamilyManager) pickVirtualGroupFamily(filter *vgmgr.PickVGFFilter, excludeSPsFilter vgmgr.ExcludeFilter) (*vgmgr.VirtualGroupFamilyMeta, error) {
 	var (
 		picker   FreeStorageSizeWeightPicker
 		familyID uint32
 		err      error
 	)
 	picker.freeStorageSizeWeightMap = make(map[uint32]float64)
 	for id, f := range vgfm.vgfIDToVgf {
-		if filter.Check(id) {
+		if filter.Check(id) && !excludeSPsFilter.Apply(id) {
```

Comment:
the Id is family id, how can it be input of a `SPsFilter`?

Reply:
should iterate GVG inside family